typing/ctype.mli

```diff
@@ -33,14 +33,52 @@ exception Matches_failure of Env.t * Errortrace.unification_error
 exception Incompatible
   (* Raised from [mcomp] *)
 
+(* Old API
 val init_def: int -> unit
         (* Set the initial variable level *)
 val begin_def: unit -> unit
         (* Raise the variable level by one at the beginning of a definition. *)
-val end_def: unit -> unit
-        (* Lower the variable level by one at the end of a definition *)
 val begin_class_def: unit -> unit
+        (* Raise the current level not touching the nongen level *)
 val raise_nongen_level: unit -> unit
+        (* Raise the nongen level to the current level *)
+val end_def: unit -> unit
+        (* Lower the variable level by one at the end of a definition *)
+*)
+
+(* All the following wrapper functions revert to the original level,
+   even in case of exception. *)
+val wrap_def: ?post:('a -> unit) -> (unit -> 'a) -> 'a
+        (* [wrap_def (fun () -> cmd) ~post] evaluates [cmd] at a raised level.
+           If given, [post] is applied to the result, at the original level.
+           It is expected to contain only level related post-processing. *)
+val wrap_def_if: bool -> (unit -> 'a) -> post:('a -> unit) -> 'a
+        (* Same as [wrap_init_def], but only raise the level conditionally.
+           [post] also is only called if the level is raised. *)
+val wrap_def_process: (unit -> 'a * 'b list) -> proc:('b -> unit) -> 'a
+        (* Variant of [wrap_def], where [proc] is iterated on the returned
+           list. *)
```

Comment:
I suggested that we call the `~proc` labels `~post` instead, for consistency. @t6s is considering renaming the combinator as well, that's fine with me.